kcidb/mq/__init__.py

```diff
@@ -546,6 +546,43 @@ def decode_data(self, message_data):
         return pattern_set
 
 
+class URLListPublisher(Publisher):
+    """URL list queue publisher"""
+
+    def encode_data(self, data):
+        """
+        Encode a list of URLs into message data.
+        Args:
+            data:   The list of URLs to encode.
+        Returns:
+            The encoded message data.
+        Raises:
+            An exception in case data encoding failed.
+        """
+        assert isinstance(data, list)
+        assert all(isinstance(url, str) for url in data)
+        return "\n".join(data).encode()
+
+
+class URLListSubscriber(Subscriber):
+    """URL list queue subscriber"""
+
+    def decode_data(self, message_data):
+        """
+        Decode message data to extract a list of URLs.
+
+        Args:
+            message_data:   The message data from the message queue
+
+        Returns:
+            The decoded list of URLs.
+
+        Raises:
+            An exception in case data decoding failed.
+        """
+        return message_data.decode().splitlines()
+
+
```

Comment:
Could you move the addition of these two classes into a separate commit, with an appropriate description? This way if we need to revert the test commit, we won't have to delete the classes. I.e. they're useful by themselves.

Reply:
Okay, moved this add from a separate branch [here](https://github.com/kernelci/kcidb/pull/454)